arangod/Cluster/MaintenanceRestHandler.cpp

```diff
@@ -185,48 +185,44 @@ void MaintenanceRestHandler::putAction() {
                     result.errorMessage());
     }  // if
   }    // if
-
-}  // MaintenanceRestHandler::putAction
+}
 
 bool MaintenanceRestHandler::parsePutBody(VPackSlice const& parameters) {
-  bool good(true);
-
   std::map<std::string, std::string> desc;
   auto prop = std::make_shared<VPackBuilder>();
   int priority = 1;
   bool forced = false;
 
   VPackObjectIterator it(parameters, true);
-  for (; it.valid() && good; ++it) {
-    VPackSlice key, value;
-
-    key = it.key();
-    value = it.value();
+  for (; it.valid(); ++it) {
+    VPackSlice key = it.key();
+    VPackSlice value = it.value();
 
-    // attempt insert into map ... but needs to be unique
     if (key.isString() && value.isString()) {
-      good = desc.insert({key.copyString(), value.copyString()}).second;
-    } else if (key.isString() && (key.stringView() == "properties") &&
+      // note: no check for key uniqueness necessary here. this
+      // is already handled when parsing the JSON request body
+      desc.insert({key.copyString(), value.copyString()});
+    } else if (key.isString() && key.stringView() == "properties" &&
                value.isObject()) {
       // code here
-      prop.reset(new VPackBuilder(value));
-    } else if (key.isString() && (key.stringView() == "priority") &&
+      prop->clear();
+      prop->add(value);
+    } else if (key.isString() && key.stringView() == "priority" &&
                value.isInteger()) {
       priority = static_cast<int>(value.getInt());
-    } else if (key.isString() && (key.stringView() == "forced") &&
+    } else if (key.isString() && key.stringView() == "forced" &&
                value.isBool()) {
       forced = value.isTrue();
     } else {
-      good = false;
-    }  // else
-  }    // for
+      return false;
```

Comment:
Is this a silent bug-fix here?
As far as I understand in original variant always an action was created.
As we could encounter an unexpected key/value pair at any position in the pinpoint the generated Action may actually be "undefined".
I prefer this variant!